questionpy/i18n.py

```diff
@@ -0,0 +1,471 @@
+
+
+def get_for(module_name: str) -> Gettext:
+    """Initializes i18n for the package owning the given Python module and returns the gettext-family functions.
+
+    Args:
+        module_name: The Python `__package__` or `__module__` whose domain should be used.
+
+    Returns: The main translation function, usually assigned to `__`. Compatible with the standard library's
+        [gettext.gettext][]. The other gettext-family functions are available as methods on `gettext`: `.n`, `.p` and
+        `.np`.
+
+    Example:
+        ```py
+        __ = i18n.get_for(__package__)
+        print(__("I'm translated!"))
+        print(__.n("One thing", "{} things", 2).format(2))
+        ```
+    """
+    package = get_package_by_python_module(module_name)
+    domain = domain_of(package.manifest)
+    domain_state = _ensure_initialized(domain, package, get_qpy_environment())
+
+    return Gettext(package, domain, domain_state)
+
+
+def dgettext(domain: str, message: str, /) -> str:
+    """Translate a message in a manually specified domain. The domain must already have been initialized."""
+    domain = GettextDomain(domain)
+    request_state = _require_request_state(domain)
+    return request_state.translations.gettext(message)
+
+
+def dpgettext(domain: str, context: str, message: str, /) -> str:
+    """Translate a message in the given context in a manually specified domain.
+
+    The domain must already have been initialized.
+    """
+    domain = GettextDomain(domain)
+    request_state = _require_request_state(domain)
+    return request_state.translations.pgettext(context, message)
+
+
+def dnpgettext(domain: str, context: str, singular: str, plural: str, n: int, /) -> str:
+    """Translate a message in the given context in a manually specified domain, accounting for plural forms.
+
+    The domain must already have been initialized.
+    """
+    domain = GettextDomain(domain)
+    request_state = _require_request_state(domain)
+    return request_state.translations.npgettext(context, singular, plural, n)
+
+
+_i18n_state: ContextVar[dict[GettextDomain, _DomainState]] = ContextVar("_i18n_state")
+
+_log = logging.getLogger(__name__)
+if _log.level == logging.NOTSET:
+    _log.setLevel(logging.INFO)
+
+_NULL_TRANSLATIONS = NullTranslations()
+
+
+def _build_translations(mos: list[Traversable]) -> NullTranslations:
+    if not mos:
+        return _NULL_TRANSLATIONS
+
+    with mos[0].open("rb") as mo_file:
+        # GNUTranslations reads the file immediately, so we can safely close it afterward.
+        translations = GNUTranslations(mo_file)
+
+    for fallback_mo in mos[1:]:
+        with fallback_mo.open("rb") as mo_file:
+            translations.add_fallback(GNUTranslations(mo_file))
+
+    return translations
+
+
+def _get_available_mos(package: Package) -> dict[Bcp47LanguageTag, Traversable]:
+    result = {}
+    locale_dir = package.get_path("locale")
+
+    for lang_dir in locale_dir.iterdir() if locale_dir.is_dir() else ():
```

Comment:
Ist eine Präferenzsache, finde aber folgendes lesbarer:

```suggestion
    if not locale_dir.is_dir():
        return {}
        
    for lang_dir in locale_dir.iterdir():
```